### DISCUSSION TITLE
[Bug]: Changes after prod deployment not possible (save=disabled)

### DISCUSSION THREAD

**Discussion by HeicoSindelItShapes** [pimcore/pimcore#15553] [Q&A / Support]
### Expected behavior

The Data-Importer-Jobs are defined in our Test-Env (env=dev).
After "deploy" the project to prod (repo-update) the configuration should be able to change (at least to change some relations)

### Actual behavior

after the deployment the job, which are created in Test-System are listed and could be opened and i can edit the fields but i can not save them because the "save"-Button is disabled

### Steps to reproduce

use the config
Test-System: APP_ENV=dev
Prod-System: APP_ENV=prod (with PIMCORE_WRITE_TARGET_DATA_HUB=settings-store) in .env

**Comment by fashxp** [comment]
related to https://pimcore.com/docs/pimcore/current/Development_Documentation/Deployment/Configuration_Environments.html

once a configuration is configured in yaml file, it cannot be updated in production environment.

**Comment by HeicoSindelItShapes** [comment]
that means i have to change for this case the PIMCORE_WRITE_TARGET_DATA_HUB from "settings-store" to "symfony-config"?

**Comment by HeicoSindelItShapes** [comment]
but at the end is explained that if i want to allow changes in Prod i have to set the TARGET... to settings-store. 
is there maybe for the data-importer a revalidation-command what i could execute as it is mentioned for the thumbnail-example
or could you provide a config, what i can use for creating jobs on non-prod, deploy them to prod and edit them there

**Comment by fashxp** [comment]
seems the docs are not clear enough yet. 

the thing is: once a configuration is stored (either as symfony-config or settings-store), the storage location doesn't change anymore for that particular configuration. the `PIMCORE_WRITE_TARGET_DATA_HUB` only applies for new configurations, not for existing ones. 

now, if you deploy a symfony-config configuration to a production system, that particular configuration cannot be changed anymore in production (as symfony-config is read-only in PROD environment and the changes would be overwritten during next deployment anyway).

if you want to allow configurations to be changed in production, you need to store their configs in settings-store. to deploy such kind of configurations, you would need to provide a migration script that adds the configuration into the settings-store table. 

the revalidation command is only feasible in combination with thumbnails - as the cached thumbnails on the production system might need to be regenerated when a changed thumbnail configuration is deployed. 


any changes in/contributions to the documentation that add additional clarity are highly appreciated - as we are already a bit biased.

**Comment by HeicoSindelItShapes** [comment]
Thank you for the explanation. I have one problem now. There is currently running a Importer which i can not change. As it is a cron-based import i am also not able to disable it there. 
Is it possible to set the PIMCORE_WRITE_TARGET_DATA_HUB=settings-store in dev-mode - so if i add a new Importer, in prod also work. As i understand it right i have to configure the import in dev with the config i will need in prod and with the deployment it will change..
